### PR TITLE
Change Security/Permission fields with overrides

### PR DESCRIPTION
**Pull request by anandswaminathan**
We want to achieve the following:

- Ability to pass IAM role information for scenarios like AWS Batch tasks
- Ability to pass service account information for python tasks.
- Ability to override permissions at task level. Different task types can use different permissions to execute in an execution.
- Deprecate the current Auth role fields 

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

### CODE REVIEW

**Comment by anandswaminathan** [comment]
@EngHabu  @katrogan 

@katrogan Where can I add for Single task execution?

**Comment by katrogan** [comment]
@anandswaminathan single task execution uses the regular CreateExecutionRequest message (the reference launch plan is actually just a task identifier instead)

File: protos/flyteidl/admin/launch_plan.proto [hunk 1]
```diff
@@ -90,7 +91,13 @@ message LaunchPlanSpec {
     // Indicates the permission associated with workflow executions triggered with this launch plan.
     Auth auth                             = 8 [deprecated=true];
 
-    AuthRole auth_role                    = 9;
+    AuthRole auth_role                    = 9 [deprecated=true];
+
+    // Indicates security context for permissions triggered with this launch plan
+    SecurityContext security_context      = 10;
+
+    // Map of tasktype to overrideable for each task type.
```
Comment (katrogan) [inline protos/flyteidl/admin/launch_plan.proto:99]: is the key value here task type name?

File: protos/flyteidl/admin/common.proto [hunk 2]
```diff
@@ -237,14 +237,46 @@ message Annotations {
...
+    // Map of custom annotations to be applied to the execution resource.
+    map<string, string> values          = 1;
+}
+
+// Config values to define privilege and access control settings
+// In the future a mode (e.g. OVERRIDE, APPEND, etc) can be defined
+// to specify how to merge configs defined at registration and execution time.
+message Config {
+    // Map of custom annotations to be applied to the execution resource.
+    map<string, string> values          = 1;
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:253]: can we call this field annotations then?

File: protos/flyteidl/admin/common.proto [hunk 3]
```diff
@@ -237,14 +237,46 @@ message Annotations {
     map<string, string> values          = 1;
 }
 
+// Security context fields to define privilege and access control settings
+// In the future a mode (e.g. OVERRIDE, APPEND, etc) can be defined
+// to specify how to merge security context defined at registration and execution time.
+message SecurityContext {
+    // Map of custom annotations to be applied to the execution resource.
+    map<string, string> values          = 1;
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:245]: can we call this field annotations?

File: protos/flyteidl/admin/launch_plan.proto (hunk 1 above)
Comment (anandswaminathan) [reply protos/flyteidl/admin/launch_plan.proto:99]: Yeah. I just followed the pattern here: https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/admin/matchable_resource.proto#L69

File: protos/flyteidl/admin/common.proto (hunk 2 above)
Comment (anandswaminathan) [reply protos/flyteidl/admin/common.proto:253]: sorry. wrong comment

File: protos/flyteidl/admin/common.proto (hunk 3 above)
Comment (anandswaminathan) [reply protos/flyteidl/admin/common.proto:245]: wrong comment.

File: protos/flyteidl/admin/common.proto [hunk 4]
```diff
@@ -237,14 +237,46 @@ message Annotations {
...
+// Deprecated
 message AuthRole {
   oneof method {
     string assumable_iam_role         = 1;
     string kubernetes_service_account = 2;
   }
 }
 
+// Defines values that can be overriden at registration/runtime
+message Overrideables {
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:266]: thoughts on naming this something more specific, like ExecutionOverrideables?

Comment (anandswaminathan) [reply protos/flyteidl/admin/common.proto:266]: Sure but it is part of LP also.

Comment (katrogan) [reply protos/flyteidl/admin/common.proto:266]: good point @EngHabu any thoughts?

**Comment by anandswaminathan** [comment]
@kumare3 @katrogan 

Discussed with @EngHabu. Overrides have a lot more scenarios that we initially anticipated. Too big to get it right here. So I am skipping, and making changes just to get off KIAM. Check next commit

**Review by EngHabu — DISMISSED** [review]

File: protos/flyteidl/admin/launch_plan.proto [hunk 5]
```diff
@@ -90,7 +91,10 @@ message LaunchPlanSpec {
     // Indicates the permission associated with workflow executions triggered with this launch plan.
     Auth auth                             = 8 [deprecated=true];
 
-    AuthRole auth_role                    = 9;
+    AuthRole auth_role                    = 9 [deprecated=true];
```
Comment (katrogan) [inline protos/flyteidl/admin/launch_plan.proto:94]: can we undeprecate this now then?

File: protos/flyteidl/admin/common.proto [hunk 6]
```diff
@@ -237,14 +237,24 @@ message Annotations {
...
+// Security context fields to define privilege and access control settings
+// In the future a mode (e.g. OVERRIDE, APPEND, etc) can be defined
+// to specify how to merge security context defined at registration and execution time.
+message SecurityContext {
+    // Map of security context values
+    map<string, string> values          = 1;
+}
+
 // Defines permissions associated with executions.
+// Deprecated
```
Comment (katrogan) [inline protos/flyteidl/admin/common.proto:249]: rm?

File: protos/flyteidl/admin/launch_plan.proto (hunk 5 above)
Comment (anandswaminathan) [reply protos/flyteidl/admin/launch_plan.proto:94]: no. why? This is deprecated

File: protos/flyteidl/admin/common.proto (hunk 6 above)
Comment (anandswaminathan) [reply protos/flyteidl/admin/common.proto:249]: why? This is getting deprecated for securityContext

File: protos/flyteidl/admin/launch_plan.proto (hunk 5 above)
Comment (katrogan) [reply protos/flyteidl/admin/launch_plan.proto:94]: oh I see so `map<string, string> values ` in SecurityContext will contain these. nvm

Comment (kumare3) [reply protos/flyteidl/admin/launch_plan.proto:94]: wow the deprecations

**Review by EngHabu — APPROVED** [review]

**Pull request merged**